File: NtunyuSerge 25 coding challenges/Exercise_16.py
#  Create a function that will return a Boolean specifying if a number is prime

def is_prime(number):
    if number <= 1:
        flag = False
    flag = False
    for i in range(2, number):  # Why is there an error when number=5 with loop running from 2 to (number//2)?
        if number % i != 0:
            flag = True
            break
    return flag


print(prime_or_not(37))

File: Randy-JRA/Exercise_16.cpp
//Challenge 16 is to Create a function that will return a Boolean specifying if a number is prime

#include <iostream>
bool is_prime(int num) {
  if (num <= 1) return false;
  for (int i = 2; i * i <= num; i++) {
    if (num % i == 0) return false;
  }
  return true;
}

int main() {
  int num;
  std::cout << "Enter a number: ";
  std::cin >> num;
  if (is_prime(num)) {
    std::cout << num << " is prime" << std::endl;
  } else {
    std::cout << num << " is not prime" << std::endl;
  }
  return 0;
}
